modeling-cmds/src/shared.rs

```diff
@@ -1062,7 +1109,33 @@ pub struct TransformBy<T> {
     pub set: bool,
     /// If true, the transform is applied in local space.
     /// If false, the transform is applied in global space.
+    #[deprecated(note = "Use the `origin` field instead.")]
     pub is_local: bool,
+    /// What to use as the origin for the transformation.
+    /// If not provided, will fall back to local or global origin, depending on
+    /// whatever the `is_local` field was set to.
+    #[serde(default)]
+    pub origin: Option<OriginType>,
+}
+
+impl<T> TransformBy<T> {
+    /// Get the origin of this transformation.
+    /// Reads from the `origin` field if it's set, otherwise
+    /// falls back to the `is_local` field.
+    pub fn get_origin(&self) -> OriginType {
+        if let Some(origin) = self.origin {
+            return origin;
+        }
+        #[expect(
+            deprecated,
+            reason = "Must fall back to the deprecated field if the API client isn't using the new field yet."
+        )]
+        if self.is_local {
+            OriginType::Local
+        } else {
+            OriginType::Global
+        }
+    }
 }
 
 impl<T: JsonSchema> JsonSchema for TransformBy<T> {
```

Comment:
If the custom impl is just about the schema name, could we derive `JsonSchema` and do this on the struct definition?

```rust
#[derive(JsonSchema)]
#[schemars(rename = "TransformByFor{T}")]
pub struct TransformBy<T> {
```

This works according to https://github.com/GREsau/schemars/issues/193#issuecomment-2944934239.